src/main/java/club/sk1er/patcher/mixins/bugfixes/optifine/RegionRenderCacheMixin_FixConnectedTextures.java

```diff
@@ -0,0 +1,36 @@
+//#if MC==10809
+package club.sk1er.patcher.mixins.bugfixes.optifine;
+
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.renderer.RegionRenderCache;
+import net.minecraft.util.BlockPos;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@SuppressWarnings("UnusedMixin") //added in PatcherMixinPlugin
+@Mixin(RegionRenderCache.class)
+public abstract class RegionRenderCacheMixin_FixConnectedTextures {
+    @Shadow
+    @Final
+    private static IBlockState DEFAULT_STATE;
+
+    @Shadow
+    protected abstract int getPositionIndex(BlockPos p_getPositionIndex_1_);
+
+    @Shadow
+    private IBlockState[] blockStates;
+
+    // yes, getPositionIndex is called twice because of this, but it doesn't really matter
```

Comment:
Rewrote the mixin so this doesn't need to be called twice, tested & works in prod.
https://hst.sh/diqojikayi.cpp